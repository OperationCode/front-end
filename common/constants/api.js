import { get, post, patch } from 'common/utils/api-utils';

/* GET REQUESTS */
export const getCodeSchoolsPromise = () => get('code_schools');
export const getMentorPromise = id => get(`mentors/${id}`);
export const getMentorsPromise = () => get('mentors');
export const getRequestsPromise = () => get('requests');
export const getScholarshipPromise = id => get(`scholarships/${id}`);
export const getScholarshipsPromise = () => get('scholarships');
export const getServicesPromise = () => get('services');
export const getTeamMembersPromise = () => get('team_members');

/* POST REQUESTS */
export const createUser = ({ email, password, firstName, lastName, zipcode }) =>
  post('users', {
    user: {
      email,
      password,
      first_name: firstName,
      last_name: lastName,
      zip: zipcode,
    },
  }).then(({ data }) => data);

export const loginUser = ({ email, password }) =>
  post('sessions', {
    user: {
      email,
      password,
    },
  }).then(({ data }) => data);

export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service }) =>
  post('requests', {
    request: {
      details: additionalDetails,
      language,
      requested_mentor_id: mentor,
      service_id: service,
    },
  });

/* PATCH REQUESTS */
export const patchUpdateMentorRequestPromise = ({ request, status, mentor }) =>
  patch(`requests/${request}`, {
    request: {
      status,
      mentor,
    },
  });
