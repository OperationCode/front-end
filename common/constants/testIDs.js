export const ALERT = 'ALERT';
export const ALERT_CLOSE_BUTTON = 'Close Alert Button';
export const BUTTON = 'BUTTON';
export const CHANGE_PASSWORD_FORM_ERROR = 'CHANGE_PASSWORD_FORM_ERROR';
export const CLOSE_BUTTON = 'CLOSE_BUTTON';
export const DOWNVOTE_BUTTON = 'DOWNVOTE_BUTTON';
export const FLAT_CARD_IMAGE = 'FLAT_CARD_IMAGE';
export const INPUT = 'INPUT';
export const INPUT_ERROR = 'INPUT_ERROR';
export const INPUT_FEEDBACK_GROUPING = 'INPUT_FEEDBACK_GROUPING';
export const LABEL = 'LABEL';
export const MULTI_STEP_SUBMIT_BUTTON = 'Submit Multi-Step Form';
export const MULTI_STEP_STEP_BUTTON = 'Submit Step Button';
export const SCREEN_READER_ONLY = 'SCREEN_READER_ONLY';
export const UPVOTE_BUTTON = 'UPVOTE_BUTTON';
