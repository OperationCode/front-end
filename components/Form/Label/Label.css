.Label {
  color: rgba(var(--rgbValuesSecondary), 0.75);
  font-size: 0.825rem;
}
