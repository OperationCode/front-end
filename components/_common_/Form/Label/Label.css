.Label {
  color: var(--secondary);
  font-size: 0.825rem;
}
