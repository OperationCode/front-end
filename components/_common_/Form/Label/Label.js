import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ScreenReaderOnly from 'components/_common_/ScreenReaderOnly/ScreenReaderOnly';
import styles from './Label.css';

Label.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  for: PropTypes.string.isRequired,
  isHidden: PropTypes.bool, // visually hides the label, but maintains accessibility
};

Label.defaultProps = {
  className: undefined,
  isHidden: false,
};

function Label({ children, className, isHidden, ...props }) {
  const TheLabel = (
    <label
      // for isnt destructured because it's a reserved word in JavaScript
      // eslint-disable-next-line react/destructuring-assignment
      htmlFor={props.for}
      className={classNames(styles.Label, className)}
    >
      {children}
    </label>
  );

  return isHidden ? <ScreenReaderOnly>{TheLabel}</ScreenReaderOnly> : <>{TheLabel}</>;
}

export default Label;
