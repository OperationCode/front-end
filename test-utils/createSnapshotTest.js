/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
