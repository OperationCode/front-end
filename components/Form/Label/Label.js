import React from 'react';
import { node, string, bool } from 'prop-types';
import classNames from 'classnames';
import { LABEL } from 'common/constants/testIDs';
import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
import styles from './Label.module.css';

Label.propTypes = {
  children: node.isRequired,
  className: string,
  'data-testid': string,
  for: string.isRequired,
  isHidden: bool, // visually hides the label, but maintains accessibility
};

Label.defaultProps = {
  className: undefined,
  'data-testid': LABEL,
  isHidden: false,
};

function Label({ children, className, 'data-testid': testID, isHidden, ...props }) {
  const TheLabel = (
    <label
      // for isnt destructured because it's a reserved word in JavaScript
      // eslint-disable-next-line react/destructuring-assignment
      htmlFor={props.for}
      className={classNames(styles.Label, className)}
      data-testid={testID}
    >
      {children}
    </label>
  );

  return isHidden ? <ScreenReaderOnly>{TheLabel}</ScreenReaderOnly> : <>{TheLabel}</>;
}

export default Label;
