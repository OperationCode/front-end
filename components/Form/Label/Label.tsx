import { node, string, bool } from 'prop-types';
import classNames from 'classnames';
import { LABEL } from 'common/constants/testIDs';
import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
import type { PropsWithChildren } from 'react';

Label.propTypes = {
  children: node.isRequired,
  className: string,
  'data-testid': string,
  for: string.isRequired,
  isHidden: bool, // visually hides the label, but maintains accessibility
};

Label.defaultProps = {
  className: undefined,
  'data-testid': LABEL,
  isHidden: false,
};

export interface LabelProps
  extends PropsWithChildren<{
    className?: string;
    ['data-testid']?: string;
    isHidden?: boolean;
    for: string;
  }> {}

function Label({ children, className, 'data-testid': testID, isHidden, ...props }: LabelProps) {
  const TheLabel = (
    <label
      // for isnt destructured because it's a reserved word in JavaScript
      // eslint-disable-next-line react/destructuring-assignment
      htmlFor={props.for}
      className={classNames('text-themeSecondary text-sm', className)}
      data-testid={testID}
    >
      {children}
    </label>
  );

  return isHidden ? <ScreenReaderOnly>{TheLabel}</ScreenReaderOnly> : TheLabel;
}

export default Label;
