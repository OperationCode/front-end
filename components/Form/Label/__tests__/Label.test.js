/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { render } from '@testing-library/react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import { LABEL, SCREEN_READER_ONLY } from 'common/constants/testIDs';

import Label from '../Label';

describe('Label', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Label for="someInputName">Test</Label>);
  });

  it('should be visually hidden with isHidden passed', () => {
    const component = render(
      <Label for="someInputName" isHidden>
        Visually Hidden?
      </Label>,
    );

    const ScreenReaderOnly = component.queryByTestId(SCREEN_READER_ONLY);
    const ActualLabelElement = component.queryByTestId(LABEL);

    expect(ScreenReaderOnly.childNodes[0]).toBe(ActualLabelElement);
  });
});
