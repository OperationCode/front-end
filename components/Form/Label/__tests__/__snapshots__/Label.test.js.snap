// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render with required props 1`] = `
<label
  className="Label"
  data-testid="LABEL"
  htmlFor="someInputName"
>
  Test
</label>
`;
